### PR TITLE
build(client): Bump build-tools dependency to 0.38

### PR DESCRIPTION
**Pull request by tylerbutler**
Updated the following:

  client (release group)

Dependencies on build-tools updated:

  @fluid-tools/build-cli: 0.38.0
  @fluidframework/build-tools: 0.38.0
  @fluidframework/bundle-size-tools: 0.38.0
  @fluid-tools/version-tools: 0.38.0

Manual fixes:

- client-utils does not have an internal entrypoint, so I updated its typetests:gen task to use the public entrypoints explicitly: `flub generate typetests --dir . -v --level public`

### CODE REVIEW

File: packages/common/client-utils/package.json [hunk 1]
```diff
@@ -90,7 +90,7 @@
 		"test:mocha:cjs": "mocha --recursive \"dist/test/mocha/**/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
 		"test:mocha:esm": "mocha --recursive \"lib/test/mocha/**/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
 		"tsc": "fluid-tsc commonjs --project ./tsconfig.cjs.json && copyfiles -f ./src/cjs/package.json ./dist",
-		"typetests:gen": "flub generate typetests --dir . -v --publicFallback",
+		"typetests:gen": "flub generate typetests --dir . -v --level public",
```
Comment (jason-ha) [inline packages/common/client-utils/package.json:93]: Can you please note in description all of the things/places that were done specially besides a version bump and build?
(My build kept failing when I used local build-tools and I was so confused as I didn't see this change in the PR.)

**Review by alexvy86 — APPROVED** [review]
Approving the devtools bit. I looked through the rest and Jason already asked all the interesting questions which we probably want answers to before merging.

File: packages/common/client-utils/package.json (hunk 1 above)
Comment (tylerbutler) [reply packages/common/client-utils/package.json:93]: Point taken. I missed it myself when reviewing the commits (I try to keep manual changes in their own commits).

File: common/lib/protocol-definitions/src/test/types/validateProtocolDefinitionsPrevious.generated.ts [hunk 2]
```diff
@@ -1680,202 +1956,12 @@ use_old_TypeAliasDeclaration_SummaryType(
...
-    get_old_VariableDeclaration_SummaryType_Attachment());
-
-/*
-* Validate back compat by using current type in place of old type
-* If breaking change required, add in package.json under typeValidation.broken:
-* "VariableDeclaration_SummaryType.Attachment": {"backCompat": false}
-*/
-declare function get_current_VariableDeclaration_SummaryType_Attachment():
-    TypeOnly<typeof current.SummaryType.Attachment>;
-declare function use_old_VariableDeclaration_SummaryType_Attachment(
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply common/lib/protocol-definitions/src/test/types/validateProtocolDefinitionsPrevious.generated.ts:1725]: I'm going to split the common package changes out from the client changes so we can discuss them/investigate individually.

[the remark this answers is not in the document]
Comment (tylerbutler) [reply common/lib/protocol-definitions/src/test/types/validateProtocolDefinitionsPrevious.generated.ts]: These changes are now in #20990.

File: packages/dds/sequence/package.json [hunk 3]
```diff
@@ -203,9 +203,6 @@
 			"ClassDeclaration_SharedSequence": {
 				"forwardCompat": false
 			},
-			"InterfaceDeclaration_ISharedString": {
-				"forwardCompat": false
-			},
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply packages/dds/sequence/package.json:208]: Investigating.

[the remark this answers is not in the document]
Comment (tylerbutler) [reply packages/dds/sequence/package.json:208]: This was a duplicate entry I removed; not sure how it was added originally.

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -11899,6 +11900,103 @@ packages:
       - aws-crt
     dev: true
 
+  /@aws-sdk/client-sso-oidc/3.569.0:
+    resolution: {integrity: sha512-u5DEjNEvRvlKKh1QLCDuQ8GIrx+OFvJFLfhorsp4oCxDylvORs+KfyKKnJAw4wYEEHyxyz9GzHD7p6a8+HLVHw==}
```
Comment (jason-ha) [inline pnpm-lock.yaml:11904]: I thought I asked about aws-sdk use and duplication in my last review, but I don't see a history of that.
Why the duplication? And per comments in a build-tools update PR, do we need them at all?

Comment (tylerbutler) [reply pnpm-lock.yaml]: I can't explain the duplication. I use the same process for updating lockfiles every time and I cannot explain why sometimes it produces some changes and other times it doesn't. I just reset it to main again and regenned it - we'll see if that's any better.

The aws-sdk dependencies and peer dependency requirements come from oclif. We don't use any of that functionality, but I don't think it's worth doing weird things to exclude the direct dependencies. For the peer dependencies, I added ignore peer dep entries in the pnpm config for @aws/* (that change wasn't in the iteration you reviewed)

File: pnpm-lock.yaml [hunk 5]
```diff
@@ -28847,6 +28925,13 @@ packages:
...
     resolution: {integrity: sha512-WMwm9LhRUo+WUaRN+vRuETqG89IgZphVSNkdFgeb6sS/E4OrDIN7t48CAewSHXc6C8lefD8KKfr5vY61brQlow==}
 
+  /ejs/3.1.10:
+    resolution: {integrity: sha512-UeJmFfOrAQS8OJWPZ4qtgHyWExa088/MtK5UEyoJGFH67cDEXkZSviOiKRCZ4Xij0zxI3JECgYs3oKx+AizQBA==}
+    engines: {node: '>=0.10.0'}
+    hasBin: true
+    dependencies:
+      jake: 10.8.7
+
   /ejs/3.1.9:
```
Comment (jason-ha) [inline pnpm-lock.yaml:28935]: Some minor unexpected "duplication" in here: @inquireer/type, @types/node, and ejs.
Maybe if we can get to newer pnpm, then we will have less of this.

**Review by jason-ha — APPROVED** [review]

**Pull request merged**